lib/spack/spack/test/concretize.py

```diff
@@ -264,6 +264,43 @@ def concretize_multi_provider(self):
         s.concretize()
         assert s['mpi'].version == ver('1.10.3')
 
+    def test_concretize_them_all(self):
```

Comment:
This test seems brittle to me:
1. It has different behaviors for the original and clingo concretizer
2. It always marks tests as passing

In general I think a test like this may be likely to enforce current behavior rather than correct behavior. I would prefer to have a single test encoding the concretization of `when-directives-true` and the diff above in `asp.py` to fix the failing test.